addons/udes_stock/models/stock_picking.py

```diff
@@ -1302,3 +1326,49 @@ def _get_suggested_location_by_packages(self, move_line_ids):
         products = mls.mapped('product_id')
 
         return self._get_suggested_location_by_products(mls, products)
+
+    def _get_suggested_location_by_height_speed(self, move_line_ids):
+        """Get location based on product height and turn over speed categories
+        """
+        Location = self.env['stock.location']
+
+        height_category = move_line_ids.mapped(
+            'product_id.u_height_category_id'
+        )
+        speed_category = move_line_ids.mapped('product_id.u_speed_category_id')
+        default_location = move_line_ids.mapped('location_dest_id')
+
+        # Check categories are constiant
+        # probably better doing a len(x) > 1 or len(y) > 1
+        # so we can return a meaningful error message
```

Comment:
Is the above a TODO msg? If so, please add the label. Otherwise, I suggest to make it more clear.